src/shogun/lib/any.h

```diff
@@ -207,7 +197,7 @@ namespace shogun
 		 */
 		virtual void set(void** storage, const void* v) const
 		{
-			*(storage) = const_cast<void*>(v);
+			*static_cast<T*>(*(storage)) = T(*reinterpret_cast<T const*>(v));
```

Comment:
party

Reply:
Funny is that I don't think it actually was correct before.